### PR TITLE
Add read the docs site

### LINKED ISSUE #96: Deploying website for each version
If you do builds of your website with e.g. [ReadTheDocs](https://about.readthedocs.com/?ref=readthedocs.com), you can make it possible for users to look at the docs at different versions of your package/deployment. For example, with input4MIPs validation, users can look at:

- [latest](https://input4mips-validation.readthedocs.io/en/latest/) i.e. the latest commit in main
- [stable](https://input4mips-validation.readthedocs.io/en/stable/) i.e. the latest tagged commit
- [v0.10.2](https://input4mips-validation.readthedocs.io/en/v0.10.2/), a specific tag
- [v0.11.3](https://input4mips-validation.readthedocs.io/en/v0.11.3/), a different specific tag

etc.

Unfortunately, with GitHub pages, this doesn't appear to be possible. [GitHub pages just always serves the data in the gh-pages branch](https://stackoverflow.com/a/37145848).

It would add more complexity (but not much, this is how we do all our docs at CR, including input4mips validation), but would this extra feature be useful? It would mean that people could easily see the state of the database/CVs at an earlier point in time, should they ever need to look at how things were at a specific commit rather than always just seeing the latest commit in main on the deployed docs site.

@durack1 interested in your thoughts here

### PR DESCRIPTION
**Pull request by znichollscr**
## Description

Closes #96.

This does not yet remove the GitHub pages build, we could do that in a follow up step if we wanted.

## Checklist

Please confirm that this pull request has done the following:

- [x] Documentation added (where applicable)
- [x] Changelog item added to `changelog/`

### CODE REVIEW

**Comment by znichollscr** [comment]
@durack1 here's what we'd need to do to build the docs on readthedocs.org, which would also allow us to close #96.

You can see a preview of the docs, as they would be built from this PR, here: https://input4mips-controlled-vocabularies-cvs--105.org.readthedocs.build/en/105/

I can add you as a maintainer to the read the docs side if you login to readthedocs.org with your GitHub account and ping me once you've done so.

The other benefit of this approach is that the docs will (attempt to) build on each PR. So, if we break the docs because of a PR, we'll find out before we merge.

File: README.md [hunk 1]
```diff
@@ -1,9 +1,12 @@
-# input4MIPs_CVs 
+<!--- --8<-- [start:description] -->
+# input4MIPs Controlled Vocabularies (CVs)
 
 [![Latest release](https://img.shields.io/badge/Latest%20release-v6.5.19-brightgreen.svg)](https://github.com/PCMDI/input4MIPs_CVs/releases/tag/v6.5.19)
 [![DOI (all versions)](https://zenodo.org/badge/doi/10.5281/zenodo.12629796.svg)](https://zenodo.org/doi/10.5281/zenodo.12629796)
+[![Docs](https://input4mips-controlled-vocabularies-cvs.readthedocs.io/en/latest/?badge=latest)](https://input4mips-controlled-vocabularies-cvs.readthedocs.io)
```
Comment (znichollscr) [inline README.md:6]: This badge won't start working until/if we merge to main

**Comment by durack1** [comment]
@znichollscr ok logged into https://readthedocs.org - with the durack1 GitHub account. Presumably that isn't required to merge this PR.. But I'll not do anything until we have stuff all hooked up, or clarification that we're good for a merge

**Comment by znichollscr** [comment]
> @znichollscr ok logged into [readthedocs.org](https://readthedocs.org) - with the durack1 GitHub account

Great, just added you

> Presumably that isn't required to merge this PR.. But I'll not do anything until we have stuff all hooked up, or clarification that we're good for a merge

Correct. Good to go if you're happy

**Pull request merged**